### PR TITLE
use typecast instead of asObservable

### LINKED ISSUE #2241: Remove and prohibit usage of `asObservable`
We are currently using `asObservable` in a lot of places when passing rxjs `Subject` or `BehaviourSubject` etc. somewhere it should not be manipulated. In the past, this already led to CPU usage issues which were fixed for example in #1405 or #2239. 
These are hard to find as they only affect the user interface when a lot of data is present, which is normally not the case within a development setup. 

The problem with `asObservable` is that it instantiates a new `Observable` object which requires computation resources, and memory allocation. It also introduces some side effects which only happen in some use cases (like in template usage - see #2239). 

As mentioned earlier the only purpose of `asObservable` is to prevent usage of write actions to a `Subject` by a consumer. Instead of creating a new `Observable` instance, we should just do a typecast of `Subject`s. 

**Example how it is done currently (bad):**
```typescript
const subject: BehaviourSubject<string[]> = new BehaviourSubject(['test'])
const observable = subject.asObservable();

// consume
observable.subscribe(...);
```

**Example how it should be done:**
```typescript
const subject: BehaviourSubject<string[]> = new BehaviourSubject(['test'])
const observable = subject as Observable<string[]>;

// consume
observable.subscribe(...);
```

### PR DESCRIPTION
**Pull request by bastianjoel**
resolves #2241

### CODE REVIEW

**Comment by jsangmeister** [comment]
What about a rule to prevent the usage in the future?

**Comment by bastianjoel** [comment]
I think we can do this with https://eslint.org/docs/latest/rules/no-restricted-properties 
Changing to draft.

**Comment by jsangmeister** [comment]
sounds good

File: client/.eslintrc.js [hunk 1]
```diff
@@ -75,6 +75,10 @@ module.exports = {
                 'simple-import-sort/imports': 'error',
                 'simple-import-sort/exports': 'error',
                 'unused-imports/no-unused-imports': 'error',
+                'no-restricted-properties': [2, {
```
[the remark this answers is not in the document]
Comment (bastianjoel) [reply client/.eslintrc.js:78]: It is the report level. Ill change that to `error` so it aligns with the rest of our rule definitions.

**Review by jsangmeister — APPROVED** [review]

**Review by luisa-beerboom — APPROVED** [review]

**Review by emanuelschuetze — APPROVED** [review]

**Pull request merged**